docs/usage.md

```diff
@@ -0,0 +1,161 @@
+# Usage
+
+## Setting the Project Root
+
+By default, the plugin considers the directory containing `mkdocs.yml` the
+project root directory. To use a different directory, specify the directory
+path in the `project_root` configuration option. The path can be absolute or
+relative to the directory containing `mkdocs.yml`.
+
+```yaml
+plugins:
+  - ... other plugin configuration ...
+  - mkdocs-autoapi:
+      project_root: path/to/project/root
+  - mkdocstrings
+```
+
+A common use case for this option is projects using the
+[src layout](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/).
+
+!!! example
+
+    Consider a project with the following structure:
+
+    ```tree
+    project/
+        .venv/
+            ...
+        docs/
+            index.md
+        src/
+            awesome_package/
+                __init__.py
+                module.py
+        tools/
+            generate_awesomeness.py
+            decrease_world_suck.py
+        mkdocs.yml
+        noxfile.py
+        pyproject.toml
+        README.md
+        setup.py
+    ```
+
+    For this project, it may or may not be desirable to include the `tools`
+    directory in the API reference and we probably don't want to include
+    the `*py` files in the top-level directory. To exclude these items, we can
+    set `project_root` to `src`:
+
+    ```yaml title="mkdocs.yml"
+    plugins:
+      - ... other plugin configuration ...
+      - mkdocs-autoapi:
+          project_root: src # or /path/to/project/src
+      - mkdocstrings
+    ```
```

Comment:
Hey @cdwilson - sorry to bug again, but I saw in the PR you reviewed recently you suggested providing documentation re: the src layout. If you don't mind, could you take a quick look at this documentation and let me know what you think? Thanks!

Reply:
Hey, just took a quick look at the docs. Overall, looks good.

Question: 

The docs say this about the `exclude` behavior:
> These patterns are evaluated relative to [project_root](https://github.com/jcayers20/mkdocs-autoapi/blob/feature-readthedocs/docs/usage.md#setting-the-project-root).

If I change the `project_root: src`, does that mean that the patterns are now evaluated relative to `src/`? (just wondering if that behavior might be confusing, since some exclude paths would need to be like `../tools/generate_awesomeness.py` instead of `tools/generate_awesomeness.py`).

Also wondering if there would be any reason for this project to have a distinction between `src` vs. `project_root` config options (i.e. the path to the directory containing the python package vs. the path to the directory containing the `mkdocs.yml` file). If so, it would allow setting the path to `src/` while keeping the exclude paths relative to the `project_root`.

Reply:
Thanks for the feedback and questions! I'll do my best to answer them :smile:

**Question 1** \
Yes, you have it right - if you set `project_root: src` then all patterns provided in `exclude` will be evaluated relative to that directory. However, in that case you would _not_ need to exclude `..tools/*.py`. The plugin only includes `*.py` files that are children or descendants of `project_dir`. Since no files in `..tools/` meet that requirement, there's no way to include them. The details of how `exclude` works are in [lines 20-56 of autoapi.py](https://github.com/jcayers20/mkdocs-autoapi/blob/develop/mkdocs_autoapi/autoapi.py#L20); the gist of it is this:

1. The universe of files to be included in the API documentation is initialized as the set of all `*.py` files that are children/descendents of `project_dir`.
2. For each pattern found in `exclude`, find all files matching the pattern relative to `project_dir` and use remove any of them from (1).

**Question 2** \
The answer above rules out the idea of keeping excluded patterns relative to the project root, I think. I'm glad you brought this up, though. Users can specify the path to `mkdocs.yml` using the `-f` [flag](https://www.mkdocs.org/user-guide/cli/) in the CLI, so I'm not sure allowing them to provide its directory as a config option provides value (it very well might; I'm still new to `mkdocs` and haven't used it much beyond simple projects).

I do, however, think in retrospect that `project_root` was a poor naming choice. I'm thinking `base_dir` might be a little more clear. I also found a bug while doing research related to this - `mkdocs` CLI commands fail when using the `-f` flag with the plugin active. I'll look into this either for 0.2.0 or soon after.

Reply:
> The plugin only includes *.py files that are children or descendants of project_dir

Got it, that clarifies my question.

> I'm thinking `base_dir` might be a little more clear.

Yeah, when I initially saw `project_root` it wasn't exactly clear what that should be. FYI, I looked to see what `sphinx-autoapi` does, and they have a config setting [`autoapi_dirs`](https://sphinx-autoapi.readthedocs.io/en/latest/reference/config.html#confval-autoapi_dirs) which takes a list of dirs to search.

> I'm still new to mkdocs and haven't used it much beyond simple projects

BTW, I'm brand new to `mkdocs` also (playing around with using it for a new project), so take everything I'm saying with a grain of salt.